### PR TITLE
Bump gradle

### PR DESCRIPTION
**Pull request by SandroMachado**
This PR bumps gradle to the latest version and updates the project to handle the breaking changes.

### CODE REVIEW

File: sample/Uphold-android-sdk-demo/app/build.gradle [hunk 1]
```diff
@@ -14,14 +14,34 @@ android {
...
 }
 
 dependencies {
-    compile fileTree(dir: 'libs', include: ['*.jar'])
-    compile 'com.android.support:appcompat-v7:22.1.1'
-    compile project (path: ':Uphold-android-sdk', configuration: 'sandboxRelease')
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.android.support:appcompat-v7:26.1.0'
+    implementation 'com.darylteo:rxjava-promises-java:1.2.1'
+    implementation 'com.squareup.retrofit:retrofit:1.9.0'
```
Comment (d-moreira) [inline sample/Uphold-android-sdk-demo/app/build.gradle]: Sort.

File: src/app/build.gradle [hunk 2]
```diff
@@ -67,20 +70,20 @@ if( android.productFlavors.size() > 0 ) {
...
-    compile 'com.darylteo:rxjava-promises-java:1.2.1'
-    compile 'com.squareup.retrofit:retrofit:1.9.0'
-    compile 'com.squareup.okhttp:okhttp:2.7.5'
-    compile 'com.squareup.okhttp:okhttp-urlconnection:2.7.5'
-    androidTestCompile 'com.android.support.test:testing-support-lib:0.1'
-    androidTestCompile 'org.hamcrest:hamcrest-library:1.3'
-    androidTestCompile ('com.github.javafaker:javafaker:0.5')
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.darylteo:rxjava-promises-java:1.2.1'
+    implementation 'com.squareup.retrofit:retrofit:1.9.0'
```
Comment (d-moreira) [inline src/app/build.gradle]: Sort.

File: src/app/build.gradle [hunk 3]
```diff
@@ -67,20 +70,20 @@ if( android.productFlavors.size() > 0 ) {
...
+    implementation 'com.squareup.okhttp:okhttp:2.7.5'
+    implementation 'com.squareup.okhttp:okhttp-urlconnection:2.7.5'
+    androidTestImplementation 'com.android.support.test:testing-support-lib:0.1'
+    androidTestImplementation 'org.hamcrest:hamcrest-library:1.3'
+    androidTestImplementation ('com.github.javafaker:javafaker:0.5')
     {
-        exclude group: 'org.yaml' //The version supported by android is at libs folder
+        exclude group: 'org.yaml' // The version supported by android is at libs folder.
```
Comment (d-moreira) [inline src/app/build.gradle]: ...`is at the libs folder.`

**Comment by SandroMachado** [comment]
@d-moreira updated.

**Pull request merged**